pkg/ring/ring.go

```diff
@@ -25,16 +25,18 @@ const (
 
 	// ConsulKey is the key under which we store the ring in consul.
 	ConsulKey = "ring"
+
+	maxExpectedReplicationSet = 5 // used to pre-allocate some arrays
 )
 
 // ReadRing represents the read inferface to the ring.
 type ReadRing interface {
 	prometheus.Collector
 
-	Get(key uint32, op Operation) (ReplicationSet, error)
-	BatchGet(keys []uint32, op Operation) ([]ReplicationSet, error)
+	Get(key uint32, op Operation, descs []IngesterDesc) (ReplicationSet, error)
```

Comment:
Can you document the new param is there to preserve allocations, and that its contents are not important - if thats true of course...